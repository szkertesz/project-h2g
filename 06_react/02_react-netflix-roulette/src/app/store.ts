import { configureStore } from '@reduxjs/toolkit';
import { apiSlice } from '../features/api/apiSlice';
// import moviesReducer from '../features/movies/moviesSlice';

export const store = configureStore({
    reducer: {
        [apiSlice.reducerPath]: apiSlice.reducer,
        // movies: moviesReducer,
    },
    middleware: getDefaultMiddleware => getDefaultMiddleware().concat(apiSlice.middleware)
})

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>