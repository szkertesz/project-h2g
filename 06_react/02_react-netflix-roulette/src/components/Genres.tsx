import GenreFilter from "./GenreFilter";
import GenreSorter from "./GenreSorter";
import classes from './Genres.module.scss'

function Genres() {
    return (
        <section className={classes.genres}>
            <GenreFilter />
            <GenreSorter />
        </section>
    );
}

export default Genres