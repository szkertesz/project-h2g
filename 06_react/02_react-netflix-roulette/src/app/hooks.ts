import {TypedUseSelectorHook, useDispatch, useSelector} from 'react-redux';
import {RootState, AppDispatch} from './store'

// https://redux.js.org/usage/usage-with-typescript#define-typed-hooks
// export const useAppDispatch: () => AppDispatch = useDispatch
export const useAppDispatch = () => useDispatch<AppDispatch>();
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;