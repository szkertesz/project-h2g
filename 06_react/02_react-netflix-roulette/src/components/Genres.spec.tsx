import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { store } from '../app/store';
import Genres from './Genres';

describe('Genres', () => {
    it('should render component', () => {
        const { container } = render(
            <Provider store={store}>
                <Genres />
            </Provider>
        );
        expect(container).toMatchSnapshot();
    });
});
