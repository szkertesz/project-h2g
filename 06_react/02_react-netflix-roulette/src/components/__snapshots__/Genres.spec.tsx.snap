// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Genres should render component 1`] = `
<div>
  <section
    class="genres"
  >
    <fieldset
      class="genre-filter"
    >
      <legend
        class="visually-hidden"
      >
        Filter Movies by Genre
      </legend>
      <div
        class="genre-filter__group"
      >
        <input
          checked=""
          class="visually-hidden"
          id="all"
          name="genre"
          type="radio"
          value=""
        />
        <label
          for="all"
        >
          All
        </label>
      </div>
      <div
        class="genre-filter__group"
      >
        <input
          class="visually-hidden"
          id="action"
          name="genre"
          type="radio"
          value="Action"
        />
        <label
          for="action"
        >
          Action
        </label>
      </div>
      <div
        class="genre-filter__group"
      >
        <input
          class="visually-hidden"
          id="comedy"
          name="genre"
          type="radio"
          value="Comedy"
        />
        <label
          for="comedy"
        >
          comedy
        </label>
      </div>
      <div
        class="genre-filter__group"
      >
        <input
          class="visually-hidden"
          data-testid="radio-input-horror"
          id="horror"
          name="genre"
          type="radio"
          value="Horror"
        />
        <label
          for="horror"
        >
          horror
        </label>
      </div>
      <div
        class="genre-filter__group"
      >
        <input
          class="visually-hidden"
          id="crime"
          name="genre"
          type="radio"
          value="Crime"
        />
        <label
          for="crime"
        >
          crime
        </label>
      </div>
      <div
        class="genre-filter__group"
      >
        <input
          class="visually-hidden"
          id="fantasy"
          name="genre"
          type="radio"
          value="fantasy"
        />
        <label
          for="fantasy"
        >
          fantasy
        </label>
      </div>
    </fieldset>
    <div
      class="genre-sorter"
    >
      <label
        for="sortingCriteria"
      >
        sort by
      </label>
      <select
        id="sortingCriteria"
        name="sortingCriteria"
      >
        <option
          selected=""
          value="releaseDate"
        >
          Release date
        </option>
        <option
          value="rating"
        >
          Rating
        </option>
        <option
          value="runtime"
        >
          Duration
        </option>
      </select>
      <fieldset>
        <input
          checked=""
          id="asc"
          name="order"
          type="radio"
          value="ascending"
        />
        <label
          for="asc"
          title="ascending order"
        >
          ↑
        </label>
        <input
          id="desc"
          name="order"
          type="radio"
          value="descending"
        />
        <label
          for="desc"
          title="descending order"
        >
          ↓
        </label>
      </fieldset>
    </div>
  </section>
</div>
`;
